packages/editor/src/store/selectors.js

```diff
@@ -1419,6 +1420,17 @@ export function getBlocksForSerialization( state ) {
 		return [];
 	}
 
+	// A single empty core/freeform block equivalent to an empty post.
+	const isSingleEmptyFreeformBlock = (
+		blocks.length === 1 &&
+		blocks[ 0 ].name === 'core/freeform' &&
+		( isEmpty( blocks[ 0 ].attributes ) || blocks[ 0 ].attributes.content === '' )
```

Comment:
Hey, I think I'm confused about this. Can you give me more( simple ) detail? 
Sorry, Gutenberg and react is still new for me so I might not be doing this right.

Reply:
Sorry for being unclear. In short: We shouldn't pretend to know anything about what attributes do or don't exist here, because it will make our lives more difficult later if we ever need to change the attributes.

It's not entirely relevant if we consider the recommendation at https://github.com/WordPress/gutenberg/pull/10404#pullrequestreview-163431784

Reply:
ok, yeah sure. Got it we should not use specific attribute name from a direct object and dependent on that. Also, this one is just checking for one empty block there might be more so it would be better if we check in all blocks in the current state. https://github.com/WordPress/gutenberg/pull/10404#pullrequestreview-163431784

I'll update this occordingly.